### PR TITLE
Added old style LoadWithAsTable API for simplifying backward compatibility.

### PR DESCRIPTION
**Pull request by sdanyliv**
@detoxhby, please review,

### CODE REVIEW

File: Source/LinqToDB/LinqExtensions.LoadWith.cs [hunk 1]
```diff
@@ -13,6 +13,50 @@ namespace LinqToDB
...
+		/// db.Table1.LoadWithAsTable(r => r.Reference1.Reference2);
+		///
+		/// // loads records from Table1 with References collection association loaded for each Table1 record
+		/// db.Table1.LoadWithAsTable(r => r.References);
+		///
+		/// // loads records from Table1 with Reference1 collection association loaded for each Table1 record
+		/// // loads records from Reference2 collection association for each loaded Reference1 record
+		/// // loads records from Reference3 association for each loaded Reference2 record
+		/// // note that a way you access collection association record (by index, using First() method) doesn't affect
+		/// // query results and allways select all records
```
Comment (detoxhby) [inline Source/LinqToDB/LinqExtensions.LoadWith.cs:36]: typ0

**Comment by detoxhby** [comment]
Another note: shouldn't the `ILoadWithQueryable` interface be directly in `LinqToDB` namespace?

Feels a bit weird using it as a member and having `LinqToDB.LinqExtensions.ILoadWithQueryable` as type (or even without `LinqToDb` root).

**Comment by sdanyliv** [comment]
Good catch, moving.

**Comment by detoxhby** [comment]
Document-side: I think there should be a warning in upgrade notes that the previous behaviour of mutating state for `ITable` is different than the new `LoadWith` as it now returns a new instance.

I know it's non-compliant way of re-using the same instance, but working code that forgot to reassign the result of `LoadWith` call will break unintentionnally after the upgrade.

**Comment by detoxhby** [comment]
Did some testing in older projects and with mimiced LoadWithAsTable works really strange.
(essentially using `LinqToDB.Linq.MethodHelper.GetMethodInfo(LinqToDB.LinqExtensions.LoadWith, table, selector)` to trick CanBuild recognition...)

There are cases where the LoadWith calls gone missing and the relations were empty even if the final Table.Expression contains the calls. It does not happen if small amount (1-2) calls were used.

I will wait for the merged version of this and retest to make sure it's not the result of my helper, but I think there should be some deeper reason why this happens (probably related to the new interface).

**Comment by sdanyliv** [comment]
What does it mean mimiced version? `LoadWithAsTable` code is fully copied from previous version.
But under hood it can be side effects for example the following code 
```cs
table.LoadWith(e => s.SomeProps.First())
```
Will return table record with only one detail record, But in 2.0 version (not checked) it will return all detail records.

Anyway `LoadWith` has three overloads:
https://github.com/linq2db/linq2db/blob/master/Source/LinqToDB/Reflection/Methods.cs#L77-L87

**Comment by detoxhby** [comment]
No, these are simple simple member access (no .First() or nested access).

By "mimicking", I did this to force the query builder do it's job even without `"LoadWithAsTable"` in CanBuild...:
```csharp
public static ITable<T> LoadWithAsTable<T>(
    this ITable<T> table,
    Expression<Func<T, object>> selector)
            where T : class
{
    if (table == null) throw new ArgumentNullException(nameof(table));

    table.Expression = Expression.Call(
        null,
        LinqToDB.Linq.MethodHelper.GetMethodInfo(LinqToDB.LinqExtensions.LoadWith, table, selector),
        new[] { table.Expression, Expression.Quote(selector) });

        return table;
    }
```
```

**Pull request merged**